rest-service/manager_rest/utils.py

```diff
@@ -411,17 +411,47 @@ def get_rrule(rule, since, until):
     if rule.get('weekdays'):
         weekdays = _get_weekdays(rule['weekdays'])
 
-    return rrule.rrule(freq=frequency, interval=interval,
-                       dtstart=since, until=until, byweekday=weekdays,
-                       count=rule.get('count'), cache=True)
-
-
-def _get_weekdays(weekdays):
-    weekdays_rule = []
-    for weekday in rrule.weekdays:
-        if str(weekday) in weekdays:
-            weekdays_rule.append(weekday)
-    return weekdays_rule
+    if not weekdays:
+        return rrule.rrule(freq=frequency, interval=interval,
+                           dtstart=since, until=until, count=rule.get('count'),
+                           cache=True)
+
+    rule_set = rrule.rruleset()
+    count = rule.get('count')
+    for weekday in weekdays:
+        rule_set.rrule(
+            rrule.rrule(freq=frequency, interval=interval, dtstart=since,
+                        until=until, byweekday=weekday[0], bysetpos=weekday[1],
+                        cache=True))
+    if not count:
+        return rule_set
+    try:
+        until_cap = rule_set[count-1]
+    except IndexError:
+        return rule_set
+    capped_rule_set = rrule.rruleset()
+    for weekday in weekdays:
+        capped_rule_set.rrule(
+            rrule.rrule(freq=frequency, interval=interval, dtstart=since,
+                        until=until_cap, byweekday=weekday[0],
+                        bysetpos=weekday[1], cache=True))
+    return capped_rule_set
+
+
+def _get_weekdays(raw_weekdays):
+    weekdays_map = {str(wd): wd for wd in rrule.weekdays}
+    weekdays = []
+    for prefixed_weekday in raw_weekdays:
+        weekday, position = prefixed_weekday[-2:], prefixed_weekday[:-2]
```

Comment:
cute

Reply:
but you already parse this input in the validation as well, I'd consider making a single function to parse it, and use it in both here, and the validation
```python
WEEKDAY_REGEXP = re.compile('(?<position>[1-4]|L-)(?<weekday>mo|tu|we|...)')  # i guess '|'.join or something, or can be just hardcoded, it's not like weekday names are ever gonna change
def _parse_complex_weekday(weekday):
   """Parse a string representing a weekday into a pair of (weekday, position)
   """
   match = WEEKDAY_REGEXP.match(weekday)
   if not match:
      return None
   return match.group('weekday'), match.group('position')  # I guess also int() or whatever
```